tools/codegen/dest/register_dispatch_key.py

```diff
@@ -86,11 +86,13 @@ def gen_create_out_helper(backend_index: BackendIndex) -> List[str]:
     return [f"""
 Tensor create_out(IntArrayRef sizes, IntArrayRef strides, const TensorOptions &options) {{
   {runtime_empty_supported_check}
+  if ({empty_options}.layout() == c10::kSparseCsr && sizes.size() == 0) {{
+      return {empty_impl}(sizes, {empty_options}.layout(c10::kStrided));
```

Comment:
Does this always make sense? It seems unlikely that if you have a CSR tensor you're always going to return a strided tensor as output for any kernel.

The situation here is similar to quantized https://github.com/pytorch/pytorch/pull/74560 but unlike with quantization I can't actually delay the choice here. So it's serious trouble for structured kernels design. I'd probably suggest not using structured as a start here.

Reply:
You want a different layout depending on the dimension. There is only 2D CSR. For a full reduction, we want a 0-dim Tensor, for a partial reduction a 1-dim COO Tensor. That only holds until we introduce dense dimensions for CSR Tensors as well.